commit 1ba61e05e254a587b7a6be9484a8bcd973551be1
Author: Zhu Yi <yi.zhu@intel.com>
Date:   Wed Feb 15 13:00:55 2006 +0800

    [PATCH] ipw2200: Fix software crypto shared WEP authentication problem
    
    We didn't set the WEP key to hardware when we are using software based
    crypto. Hardware needs the key to do WEP authentication even for
    software based encryption.
    
    Signed-off-by: Zhu Yi <yi.zhu@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ipw2200.c b/drivers/net/wireless/ipw2200.c
index 02d2dae85c4c..30b13a6e6fd9 100644
--- a/drivers/net/wireless/ipw2200.c
+++ b/drivers/net/wireless/ipw2200.c
@@ -7061,8 +7061,7 @@ static int ipw_associate_network(struct ipw_priv *priv,
 		priv->assoc_request.auth_type = AUTH_SHARED_KEY;
 		priv->assoc_request.auth_key = priv->ieee->sec.active_key;
 
-		if ((priv->ieee->sec.level == SEC_LEVEL_1) &&
-		    !(priv->ieee->host_encrypt || priv->ieee->host_decrypt))
+		if (priv->ieee->sec.level == SEC_LEVEL_1)
 			ipw_send_wep_keys(priv, DCW_WEP_KEY_SEC_TYPE_WEP);
 
 	} else if ((priv->capability & CAP_PRIVACY_ON) &&